### PR TITLE
[BACKLOG-4030] - Review Component - Date Picker

### PR DESCRIPTION
**Pull request by plagoa**
- Allow default null or empty specified value;
- Allow change of date formats in the date component;
- Some code refactoring;

### CODE REVIEW

**Comment by plagoa** [comment]
@diogofscmariano @annapopova1 @krivera-pentaho @scottyaslan @rfellows @pamval please review

File: package-res/resources/web/prompting/components/DojoDateTextBoxComponent.js [hunk 1]
```diff
@@ -89,16 +89,21 @@ define(['cdf/components/BaseComponent', "dojo/date/locale", 'dijit/form/DateText
           var parameterValue = this.dashboard.getParameterValue(this.parameter),
               value = undefined;
 
-          if(parameterValue.length == 1) {
+          // If no value was specified use current date
+          if (!parameterValue.length){
```
Comment (krivera-pentaho) [inline package-res/resources/web/prompting/components/DojoDateTextBoxComponent.js]: Why wouldn't you just use  [[ if (!parameterValue) { ]]. Is or can parameterValue be null?

Comment (plagoa) [inline package-res/resources/web/prompting/components/DojoDateTextBoxComponent.js]: parameterValue wil return an empty array if no value is specified when creating the prompt, so I have to check for its length.

**Comment by annapopova1** [comment]
Is really needed "_isLegacyDateFormat" and "_convertFormat" functions?
Sorry, but I can't understand why we compare jquery and dojo formatting if now we are using dojo implementation...

**Comment by pamval** [comment]
That was introduced for backward compatibility. If an user opens a dashboard with date picker prompts saved in an earlier version of dashboards (5.4), that logic makes sure the format doesn't break and the dashboard is still usable.

At least that was the goal

**Comment by annapopova1** [comment]
Probably would be useful add comments to code about it. It helps do not forget about this fact for different developers.

**Comment by pamval** [comment]
@plagoa, can you add the comments Anna is mentioning ? I agree it makes a lot of sense to have them.

**Comment by plagoa** [comment]
Updated documentation and made a minor change to the code.

@diogofscmariano @annapopova1 @krivera-pentaho @scottyaslan @rfellows @pamval please review

**Pull request merged**